doc/decisions/error_message.md

```diff
@@ -0,0 +1,137 @@
+#Error message & handling concept
+
+## Problem
+
+The current error concept has disadvantages in following regards:
+
+- Too verbose error message
+- A lot of redundant errors
+- Hard to manage specification file
+- No senseful way for application developers to use error codes from elektra
+
+This concept should improve useability for end users and developers.
+
+## Constraints
+
+- Error numbers must stay because they are more reliable to check against than strings
```

Comment:
There are many more constraints, e.g.:

- supporting multiple programming languages
- constraints from the plugin system
- ...

Reply:
How does it exactly differ when supporting multiple programming languages?
What does the plugin system exactly constraint in this case?

Reply:
> How does it exactly differ when supporting multiple programming languages?

The errors should be in a way so that the specification can be mapped to exceptions (or other ways different programming languages handle errors).

> What does the plugin system exactly constraint in this case?

In many ways, for example, multiple plugins might try to set an error and the system needs to decide which is the error to be shown to the user. Hopefully you learn such constraints from related work.